app/assets/javascripts/discourse/tests/unit/routes/review-index-test.js

```diff
@@ -0,0 +1,66 @@
+import User from "discourse/models/user";
+import { setupTest } from "ember-qunit";
+import { module, test } from "qunit";
+
+module("Unit | Route | review-index", function (hooks) {
+  setupTest(hooks);
+
+  test("subscribes and unsubscribes /reviewable_counts when user menu not enabled", function (assert) {
+    const currentUser = User.create({ redesigned_user_menu_enabled: false });
+    this.owner.unregister("service:current-user");
+    this.owner.register("service:current-user", currentUser, {
+      instantiate: false,
+    });
+    this.owner.inject("route", "currentUser", "service:current-user");
```

Comment:
Just being careful here because I'm not 100% sure what side effects this might have. Does `this.owner` refer to some global state where altering the state might result in a leak to another test?

Reply:
Currently in every before-each for `setupRenderingTests` where this service is used, it's being reset every time:

https://github.com/discourse/discourse/blob/e3f1e0e9bcf89d960cc13cfa1bc56f60a89055a3/app/assets/javascripts/discourse/tests/helpers/component-test.js#L12-L50

I don't believe this style of injecting `currentUser` is present in most unit tests. But anyway I duplicated a single test and can confirm that there's no pollution.